### ISSUE TITLE
RKE2 Provisioning: Cluster Options - Kubernetes Version - 1.21 Only

### ISSUE THREAD

**Issue by deniseschannon** [rancher/rancher#32364]
Confirm that only k8s 1.21+ is shown when RKE2 provisioning feature flag is on. 

* Node Provisioning
* Custom
* Imported RKE2 clusters  
 
Test importing a k8s 1.19 cluster to see what happens with available 

Test use cases where RKE2 provisioning feature flag is on vs. off

**Comment by deniseschannon** [comment]
```
apiVersion: provisioning.cattle.io/v1
kind: Cluster
metadata:
  name: #string
  annotations:
    {}
#    key: string
  labels:
    {}
#    key: string
  namespace: fleet-default
spec:
  cloudCredentialSecretName: do
  kubernetesVersion: v1.20.5+rke2r1
```

**Comment by deniseschannon** [comment]
NOTE: Currently the options are available for 1.20 and 1.21 in order to test downstream cluster upgrades.

**Comment by thehejik** [comment]
**Test report:**

Tested on `rancher/rancher:master-head c7f3cce`

**Confirm that only k8s 1.21+ is shown when RKE2 provisioning feature flag is on.**
* For RKE2 clusters (deployable from Vue only) I see there is currently only one rke2 k8s version available in UI `v1.21.2+rke2r1`

**Node Provisioning
Custom**
* Tested within https://github.com/rancher/rancher/issues/32374 and https://github.com/rancher/dashboard/issues/2736 (for the cloud node drivers mainly UI was tested)

**Imported RKE2 clusters
Test importing a k8s 1.19 cluster to see what happens with available**
* I was able test the upgrade of imported RKE2 cluster but only in Ember UI (due an issue below) - I did successful upgrade from `v1.20.8+rke2r1` to  `v1.21.2+rke2r1` - system resources were ok
* I was testing the import mainly in Vue but in Ember as well - the reg. command in both UIs is using very similar registration command and `yaml`(~ generated values differ) - problem in Vue is that when importing the cluster it doesn't belong to any Owner and user is unable to upgrade imported RKE2 from Vue (not sure if these issues are connected) - see issues below:

  - **NEW** https://github.com/rancher/dashboard/issues/3337
  - **NEW** https://github.com/rancher/dashboard/issues/3338
  - **EXISTING** https://github.com/rancher/dashboard/issues/3183 ~ Imported RKE2 cluster from Vue doesn't belong to any Owner

**Test use cases where RKE2 provisioning feature flag is on vs. off**
* Tested within https://github.com/rancher/rancher/issues/32374

Closing as bugs are opened and other than that it works as expected.
